### PR TITLE
Closes #599: XSettings handling on UI

### LINKED ISSUE #599: XSettings handling on UI
Final ui form model, converter, parameters binding.

### PR DESCRIPTION
**Pull request by lukdumi**

### CODE REVIEW

File: saos-common/src/main/java/pl/edu/icm/saos/common/chart/value/Week.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    
+    
+    
+    //------------------------ CONSTRUCTORS --------------------------
+    
+    /**
+     * 
+     * @param startDate {@link #getStartDate()}, must not be null and must be before end date (no more
+     * than 6 days)
+     * @param endDate {@link #getEndDate(), must not be null and must be after the start date (no more than 6 days)
```
Comment (madryk) [inline saos-common/src/main/java/pl/edu/icm/saos/common/chart/value/Week.java]: lack of closing parenthesis in `@link`

File: saos-restful-web-api/src/main/java/pl/edu/icm/saos/api/analysis/SeriesConverter.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+import pl.edu.icm.saos.api.analysis.ApiChart.ApiSeries;
+import pl.edu.icm.saos.common.chart.Point;
+import pl.edu.icm.saos.common.chart.Series;
+import pl.edu.icm.saos.common.chart.formatter.PointFormatter;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * A {@link Series} to {ApiSeries} converter.
```
Comment (madryk) [inline saos-restful-web-api/src/main/java/pl/edu/icm/saos/api/analysis/SeriesConverter.java]: `{@link ApiSeries}` ?

File: saos-restful-web-api/src/main/java/pl/edu/icm/saos/api/analysis/SeriesConverter.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    //------------------------ LOGIC --------------------------
+    
+    /**
+     * Converts {@link Series} into {@link ApiSeries} 
+     */
+    public ApiSeries convert(Series<?, ?> series) {
+        
+        Preconditions.checkNotNull(series);
+        
+        ApiSeries uiSeries = new ApiSeries();
```
Comment (madryk) [inline saos-restful-web-api/src/main/java/pl/edu/icm/saos/api/analysis/SeriesConverter.java]: uiSeries -> apiSeries ?

File: saos-restful-web-api/src/main/java/pl/edu/icm/saos/api/analysis/SeriesConverter.java (hunk 2 above)
Comment (lukdumi) [inline saos-restful-web-api/src/main/java/pl/edu/icm/saos/api/analysis/SeriesConverter.java]: :+1:

File: saos-restful-web-api/src/main/java/pl/edu/icm/saos/api/analysis/SeriesConverter.java (hunk 3 above)
Comment (lukdumi) [inline saos-restful-web-api/src/main/java/pl/edu/icm/saos/api/analysis/SeriesConverter.java]: :+1:

File: saos-search/src/test/java/pl/edu/icm/saos/search/analysis/solr/result/FacetDateValueConverterTest.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+    
+    @Test
+    public void convert() {
+        
+        // given
+        String value = "2010-12-07T00:00:00Z";
+        
+        // execute
+        assertEquals(new LocalDate(2010, 12, 07), converter.convert(value));
```
Comment (madryk) [inline saos-search/src/test/java/pl/edu/icm/saos/search/analysis/solr/result/FacetDateValueConverterTest.java]: Not an issue. But for me is a little weird to see here 7 in octal numeral system

Comment (lukdumi) [inline saos-search/src/test/java/pl/edu/icm/saos/search/analysis/solr/result/FacetDateValueConverterTest.java]: :+1:

File: saos-webapp/src/main/java/pl/edu/icm/saos/webapp/analysis/UiAnalysisConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    
+    @Bean
+    public PointValueFormatter pointWeekBrValueFormatterManager() {
+        return new PointWeekBrValueFormatter();
+    }
+    
+    @Bean
+    public PointValueFormatter pointObjectValueFormatterManager() {
+        return new PointObjectValueFormatter();
+    }
```
Comment (madryk) [inline saos-webapp/src/main/java/pl/edu/icm/saos/webapp/analysis/UiAnalysisConfiguration.java]: pointObjectValueFormatterManager -> pointObjectValueFormatter and analogical for beans defined above?

File: saos-webapp/src/main/java/pl/edu/icm/saos/webapp/analysis/UiAnalysisService.java [hunk 6]
```diff
@@ -37,16 +38,16 @@
...
-    private ChartConverter chartConverter;
+    private ChartConverter flotChartConverter;
     
     
     
     //------------------------ LOGIC --------------------------
     
     /**
-     * Generates and returns a {@link UiChart} according to the settings in the passed analysisForm  
+     * Generates and returns a {@link ApiChart} according to the settings in the passed analysisForm
```
Comment (madryk) [inline saos-webapp/src/main/java/pl/edu/icm/saos/webapp/analysis/UiAnalysisService.java]: `{@link ApiChart}` -> `{@link FlotChart}`

File: saos-webapp/src/main/java/pl/edu/icm/saos/webapp/analysis/UiAnalysisConfiguration.java (hunk 5 above)
Comment (lukdumi) [inline saos-webapp/src/main/java/pl/edu/icm/saos/webapp/analysis/UiAnalysisConfiguration.java]: :+1:

File: saos-webapp/src/main/java/pl/edu/icm/saos/webapp/analysis/UiAnalysisService.java (hunk 6 above)
Comment (lukdumi) [inline saos-webapp/src/main/java/pl/edu/icm/saos/webapp/analysis/UiAnalysisService.java]: :+1:

File: saos-common/src/main/java/pl/edu/icm/saos/common/chart/formatter/PointWeekValueFormatter.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+import com.google.common.base.Preconditions;
+
+/**
+ * 
+ * {@link PointValueFormatter} implementation handling {@link Week} class
+ * 
+ * @author Łukasz Dumiszewski
+ */
+public class PointWeekValueFormatter extends AbstractPointValueFormatter {
```
Comment (madryk) [inline saos-common/src/main/java/pl/edu/icm/saos/common/chart/formatter/PointWeekValueFormatter.java]: This class is not used anywhere. Do you plan to use it in future?

Comment (lukdumi) [inline saos-common/src/main/java/pl/edu/icm/saos/common/chart/formatter/PointWeekValueFormatter.java]: First, I wanted to use it on ui, then I realized I needed another one (with br). I left it for the sake of rest api.

**Pull request closed without merging**